### PR TITLE
Integer overflow 4516 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Avoids some warnings about integers

Obviously not complete yet... But still mergeable ?

libhtp-pr: 338

### CODE REVIEW

File: src/app-layer-detect-proto.c [hunk 1]
```diff
@@ -72,11 +72,10 @@ typedef struct AppLayerProtoDetectProbingParserElement_ {
     uint16_t port;
     /* \todo calculate at runtime and get rid of this var */
     uint32_t alproto_mask;
-    /* \todo check if we can reduce the bottom 2 vars to uint16_t */
```
Comment (catenacyber) [inline src/app-layer-detect-proto.c:75]: yes we can

File: src/app-layer-detect-proto.c [hunk 2]
```diff
@@ -319,7 +319,7 @@ static inline int PMGetProtoInspect(AppLayerProtoDetectThreadCtx *tctx,
  *  \param direction direction for the patterns
  *  \param pm_results[out] AppProto array of size ALPROTO_MAX */
 static AppProto AppLayerProtoDetectPMGetProto(AppLayerProtoDetectThreadCtx *tctx, Flow *f,
-        const uint8_t *buf, uint16_t buflen, uint8_t flags, AppProto *pm_results, bool *rflow)
+        const uint8_t *buf, uint32_t buflen, uint8_t flags, AppProto *pm_results, bool *rflow)
```
Comment (catenacyber) [inline src/app-layer-detect-proto.c:322]: I did not manage to get `buflen` bugger than `UINT16_MAX` here, with big TCP window, first out of order packet, then first packet, with both of them getting a bigger payload than UINT16_MAX...
Is is possible ?

File: src/app-layer-htp.c [hunk 3]
```diff
@@ -2404,14 +2404,12 @@ static void HTPConfigSetDefaultsPhase2(const char *name, HTPCfgRec *cfg_prec)
         int rdrange = cfg_prec->randomize_range;
 
         long int r = RandomGetWrap();
-        cfg_prec->request.inspect_min_size +=
```
Comment (catenacyber) [inline src/app-layer-htp.c:2407]: Do people use these randomize stuff ?

File: src/source-pcap-file.c [hunk 4]
```diff
@@ -412,14 +410,6 @@ static TmEcode DecodePcapFile(ThreadVars *tv, Packet *p, void *data)
     /* update counters */
     DecodeUpdatePacketCounters(tv, dtv, p);
 
-    double curr_ts = p->ts.tv_sec + p->ts.tv_usec / 1000.0;
-    if (curr_ts < prev_signaled_ts || (curr_ts - prev_signaled_ts) > 60.0) {
-        prev_signaled_ts = curr_ts;
```
Comment (catenacyber) [inline src/source-pcap-file.c:417]: Could we not have a warning `prev_signaled_ts` is set but unused ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 4668

File: src/app-layer-detect-proto.c (hunk 2 above)
Comment (victorjulien) [reply src/app-layer-detect-proto.c:322]: I think in theory it might be. Debug validation bug on to see if fuzzing finds a case?

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6549

File: src/app-layer-detect-proto.c (hunk 2 above)
Comment (catenacyber) [reply src/app-layer-detect-proto.c:322]: Trying that...

Comment (catenacyber) [reply src/app-layer-detect-proto.c:322]: So, CIFuzz found cases in next version ;-)

**Pull request closed without merging**